File: app/src/main/java/com/imangazalievm/bubbble/di/UserShotsPresenterComponent.java
package com.imangazalievm.bubbble.di;

import com.imangazalievm.bubbble.di.modules.UserShotsPresenterModule;
import com.imangazalievm.bubbble.di.scopes.Presenter;
import com.imangazalievm.bubbble.presentation.mvp.presenters.UserProfilePresenter;
import com.imangazalievm.bubbble.presentation.mvp.presenters.UserShotsPresenter;

import dagger.Component;

@Presenter
@Component(dependencies = ApplicationComponent.class, modules = UserShotsPresenterModule.class)
public interface UserShotsPresenterComponent {

    UserShotsPresenter getPresenter();

}

File: app/src/main/java/com/imangazalievm/bubbble/di/userprofile/UserShotsPresenterComponent.kt
package com.imangazalievm.bubbble.di.userprofile

import com.imangazalievm.bubbble.di.global.scopes.Presenter
import com.imangazalievm.bubbble.di.global.ApplicationComponent
import com.imangazalievm.bubbble.presentation.userprofile.shots.UserShotsPresenter
import dagger.Component

@Presenter
@Component(
    dependencies = [ApplicationComponent::class],
    modules = [UserShotsPresenterModule::class]
)
interface UserShotsPresenterComponent {

    fun getPresenter(): UserShotsPresenter

}